lib/internal/Magento/Framework/Model/ResourceModel/Db/AbstractDb.php

```diff
@@ -605,6 +605,7 @@ protected function _checkUnique(\Magento\Framework\Model\AbstractModel $object)
         if (!empty($fields)) {
             if (!is_array($fields)) {
                 $this->_uniqueFields = [['field' => $fields, 'title' => $fields]];
+                $fields = $this->_uniqueFields;
```

Comment:
Please combine in single assignment.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉